### PR TITLE
refactor: disallow kwargs for `Flag` subclasses

### PR DESCRIPTION
**Pull request by spifory**
This pull request explicitly types parameters in `Flag` subclasses such as UserFlags, Permissions, Intents, and MessageFlags.

This has not been fully tested; It does not look like any problem can occur with this, however better safe than sorry.

### CODE REVIEW

**Review by EmmmaTech — APPROVED** [review]
It looks good, however I would like to use a LibCST codemod in the future to achieve this (similar to Disnake).

**Comment by EmmmaTech** [comment]
I don't see how this is a breaking change because it's only typing the `__init__` parameters during type checking.

File: discatpy/bot.py [hunk 1]
```diff
@@ -15,6 +15,31 @@
 
 
 class Intents(Flag):
+    if t.TYPE_CHECKING:
+        def __init__(
+            self,
+            guilds: bool = ...,
```
Comment (EmmmaTech) [inline discatpy/bot.py:23]: These parameters are not keyword forced.

File: discatpy/models/message.py [hunk 2]
```diff
@@ -109,6 +109,20 @@ class MessageTypes(int, Enum):
 
 
 class MessageFlags(Flag):
+    if t.TYPE_CHECKING:
+        def __init__(
+            self,
+            crossposted: bool = ...,
```
Comment (EmmmaTech) [inline discatpy/models/message.py:117]: Same here

File: discatpy/models/permissions.py [hunk 3]
```diff
@@ -8,6 +8,51 @@
 
 
 class Permissions(Flag):
+    if t.TYPE_CHECKING:
+        def __init__(
+            self,
+            create_instant_invite: bool = ...,
```
Comment (EmmmaTech) [inline discatpy/models/permissions.py:17]: And here

File: discatpy/models/user.py [hunk 4]
```diff
@@ -30,6 +30,26 @@ class UserPremiumTypes(int, Enum):
 
 
 class UserFlags(Flag):
+    if t.TYPE_CHECKING:
+        def __init__(
+            self,
+            staff: bool = ...,
```
Comment (EmmmaTech) [inline discatpy/models/user.py:38]: You get the point

**Review by EmmmaTech — CHANGES_REQUESTED** [review]

**Comment by spifory** [comment]
> I don't see how this is a breaking change because it's only typing the `__init__` parameters during type checking.

Good point, the title change would be appropriate then.

File: discatpy/models/user.py (hunk 4 above)
Comment (spifory) [reply discatpy/models/user.py:38]: [`b9449ce` (#18)](https://github.com/discatpy-dev/library/pull/18/commits/b9449ce65dede1ffe8bd714334dfb47f61533a39)

File: discatpy/models/permissions.py (hunk 3 above)
Comment (spifory) [reply discatpy/models/permissions.py:17]: [`b9449ce` (#18)](https://github.com/discatpy-dev/library/pull/18/commits/b9449ce65dede1ffe8bd714334dfb47f61533a39)

File: discatpy/models/message.py (hunk 2 above)
Comment (spifory) [reply discatpy/models/message.py:117]: [`b9449ce` (#18)](https://github.com/discatpy-dev/library/pull/18/commits/b9449ce65dede1ffe8bd714334dfb47f61533a39)

File: discatpy/bot.py (hunk 1 above)
Comment (spifory) [reply discatpy/bot.py:23]: [`b9449ce` (#18)](https://github.com/discatpy-dev/library/pull/18/commits/b9449ce65dede1ffe8bd714334dfb47f61533a39)

**Review by EmmmaTech — APPROVED** [review]
LGTM

**Pull request merged**